src/Core/Domain/Product/VirtualProductFile/Exception/VirtualProductFileConstraintException.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\VirtualProductFile\Exception;
+
+/**
+ * Thrown when virtual product file constraints are violated
+ * Each constant represents violated property
+ */
+class VirtualProductFileConstraintException extends VirtualProductFileException
+{
+    //@todo: check https://github.com/PrestaShop/PrestaShop/pull/21110#discussion_r519180880
```

Comment:
So from the discussion in PR 21110 seems like we should validate all properties
I'm not sure what you meant by `In this case, maybe it's not worth adding constants (error codes) in constraint exception for such properties ?`

Do you mean we should validate them but without a dedicated error code for each property? Then how would be know which field is invalid?

Reply:
yeah... Anyway, lets just keep it simple and check every prop. (and have error codes for it)